### PR TITLE
Update exchanges category selection

### PR DESCRIPTION
**Pull request by Jujhar**
Addresses #674 

Changed a few things relating to the selection filtration box.

Screen Shot 2019-06-06 at 5 46 21 PM

Added labels based on currently selected button (like High Liquidity) and links to different sections of the page. (like OTC Trading)
Screen Shot 2019-06-08 at 4 18 06 PM

You can now select a particular exchange through the anchor tag
http://localhost:1313/exchanges/#Bittrex

Screen Shot 2019-06-08 at 4 23 15 PM

Moving to the central top location emphasizes tool not sure if more textual labels are required then to gauge the user to use buttons.

### CODE REVIEW

**Comment by linnutee** [comment]
Could you share a hosted preview for this, so it could be tested?

**Comment by linnutee** [comment]
Exchanges OTC Traders and Payment Processors are categories, so extending the filtering section to those isn’t optimal as the last two don’t have any filters + creates an odd-out new component.

Screenshot 2019-06-13 at 18 51 15

In order to keep consistency this can be solved with the tabs, similar as used in contributors subpage and news subpage (not yet merged). Showing a mockup here, though stick with the current styling of the tabs, best if those get updated in one go. 

Screenshot 2019-06-13 at 19 10 28

Filter option (e.g. high liquidity) behind the title this way seems fine.

**Comment by Jujhar** [comment]
Looks good will implement shortly.

**Comment by Jujhar** [comment]
> Could you share a hosted preview for this, so it could be tested?

It's actually very easy to setup.

All it requires mainly is hugo and one or two files that come up as errors such as jt, the translation files, and one other one, so just install hugo and when an error comes like jt not installed  or jt error or something like that go and install jt as well through command line. Try it a little.

**Comment by dajohi** [comment]
@Jujhar Please join the Decred Matrix chat community.  See https://decred.org/matrix

**Comment by Jujhar** [comment]
Updated 
exchanges

**Comment by linnutee** [comment]
Some noticeable issues still. 

1) As mentioned earlier, use the exact same styling for tabs as present in the /contributors subpage. 
2) The alignment of the cards and the selector is offset. 
3) Show all uses an out-of-scheme gray color instead of `#8997A5` and`#2970FF`
4) Spacing above and below the tabs should be significantly reduced. Eg. 30px steps. 

Screenshot 2019-07-02 at 22 28 55

**Comment by Jujhar** [comment]
Sorry missed the note about the contributors page.
Fixed it now. 

exchanges-payments

**Comment by linnutee** [comment]
OTC Traders and Payment processors seem to have an out of scheme color value `#8e8e8e`
The spacing looks way too small now, should be 30px from top and either 30px or 50px between tabs and the title.

**Comment by Jujhar** [comment]
Updated per request.

**Comment by peterzen** [comment]
Hosted copy for testing: https://dcrweb.netlify.com

**Comment by peterzen** [comment]
> Updated per request.

The category selection menu items seem hanging in the air, and they don't look like clickable links.

image

**Comment by linnutee** [comment]
Thanks, the netlify preview helps. Could you just use the exact same UI pattern for tabs from contributors subpage? Right now this seems to be made from scratch and is visually off – so we would end up with two different instances of the exact same pattern.

Also spacing is still off, too tight.

**Comment by Jujhar** [comment]
Updated, here is screen

Screen Shot 2019-07-23 at 4 45 49 PM

**Comment by linnutee** [comment]
Screenshot 2019-07-29 at 17 54 35

Quick glance off screenshot, this looks about quite close – but slightly different when comparing purely off screenshots. As I can't tell whether this may be due to the screenshot, could another netlify preview be hosted also?
 
Do they also work the same way as the tabs in contractors subpage or are they with a "scroll to" interaction as in the previous netlify preview?

The spacing still has issues (both the steps between titles and distance between filtering box + cards), however these can pass atm as i'd rather see a global spacing fix done across the site once the breadcrumbs nav few new subpages are out.

**Comment by peterzen** [comment]
The netlify preview is not updated: https://dcrweb.netlify.com/

**Comment by Jujhar** [comment]
They are using the same CSS but I guess the font can be increased decreased. 
These are 'scroll to' at the moment should I make it flash in flash out?

**Comment by linnutee** [comment]
Ok if it's the same css, keep it as it is. Probably difference from the screenshot then. 

They should function the same way as the tabs in the contributors subpag ,as otherwise the vertical navigation isn't improved.

**Comment by Jujhar** [comment]
Removed 'Scroll to' navigation. 

I had to add back the replicate css clone class instead of directly linking to because using that same class broke they dynamic filter selection. 

I think this is ready to merge.

**Comment by linnutee** [comment]
Could use a preview for this.

**Comment by linnutee** [comment]
Looks good, cheers.

File: src/assets/js/main-ex.js [hunk 1]
```diff
@@ -1,5 +1,42 @@
 $(function(){
 
+    // Retrieve category by selected get params
```
Comment (jholdstock) [inline src/assets/js/main-ex.js]: Alot of unnecessary duplication in this js code. Clean this up please.

I don't think we should use `console.log` in poduction either

File: src/assets/js/main-ex.js [hunk 2]
```diff
@@ -143,4 +202,67 @@ $(function(){
           }
        })
     })
-});
\ No newline at end of file
+});
+
+function exchangesClick(){
```
Comment (jholdstock) [inline src/assets/js/main-ex.js]: Again, a lot of unnecessary duplicated code here. This needs to be cleaned up

File: transifex_catalogs/en.json [hunk 3]
```diff
@@ -67,7 +67,10 @@
   "exchanges_payment_title": "Decred Payment Processors",
   "exchanges_payment_description": "Solutions for online merchants to start accepting Decred (DCR) as payment. Payment processors provide a quick and easy integration to accept cryptocurrencies through e-commerce platforms.",
   "exchanges_subtitle": "Decred on Exchanges",
-  "exchanges_showall": "Show all",
+  "exchanges_showall": "Exchanges",
```
Comment (jholdstock) [inline transifex_catalogs/en.json]: I may be incorrect on this, but I believe these new keys needed to be added to transifex files for all languages, not just `en`. @peterzen could you please confirm?

File: src/assets/js/main-ex.js [hunk 4]
```diff
@@ -53,6 +90,26 @@ $(function(){
...
+        // Display selected filter in exchanges-filter subheader neatly
+        var output = ""
+        if (group !== "all") {
+
+          // Add selected param to url
+          $(this).attr('href', function() {
+
+              // Do not repeat
+              if (this.href.includes(group.toString())) {
+                console.log("Same category clicked twice.");
```
Comment (jholdstock) [inline src/assets/js/main-ex.js]: Remove console.log

File: transifex_catalogs/en.json (hunk 3 above)
Comment (peterzen) [reply transifex_catalogs/en.json]: It's correct as it is - the other languages will get populated after they've been translated in Transifex. This file will have to be pushed to TF once the PR is approved, using `bin/transifex_push.sh`

**Comment by peterzen** [comment]
> 2. URL format has changed. Previously:
> 
> ```
> https://decred.org/exchanges/#high-liquidity
> ```
> 
> New
> 
> ```
> https://decred.org/exchanges/#&a=high-liquidity
> ```
> 
I'm afraid an earlier change (in #711 ) conflicts with this PR, causing the URL format to change.

I believe the correct way to implement this would be to render the 3 different sections using different Hugo templates (instead of the current JS based approach).   `src/layouts/partials/exchanges-grid.html` which currently renders all 3 is way too long anyway, this could be broken up into 3 layouts, each rendering just one section.

That way the URL structure would change to `/exchanges/otc`, `/exchanges/payment`, `/exchanges/` (seeing this spelled out, this needs some thought and possibly a compatibility redirect), and the filter on the Exchanges grid would keep using the current URL hash.

Thoughts?

**Comment by Jujhar** [comment]
The URLs are all working perfectly using #high-liquidity including buttons and without JS 
https://compassionate-aryabhata-7d8d94.netlify.com/exchanges/

File: transifex_catalogs/en.json (hunk 3 above)
Comment (Jujhar) [reply transifex_catalogs/en.json]: Ok

**Comment by Jujhar** [comment]
> Can you please squash the commits in the PR?

Updated and ready to merge I think.

**Comment by Jujhar** [comment]
Been thoroughly updated and tested. 
https://compassionate-aryabhata-7d8d94.netlify.com/exchanges/

Curious about the !important thing causing an issue do you have a link so I can weigh it accordingly. 
What’s the best way you think to proceed with this—I am open to changing it to non magic numbers?

**Comment by Jujhar** [comment]
Hi

Did check this?
Not sure what you think cause usually a week or two response.

**Comment by noahpierau** [comment]
https://compassionate-aryabhata-7d8d94.netlify.com/exchanges/ 
Is this the latest test version of the website? 

Two remarks based on this version.

First, examine at the URL structure. Right now the URLs are:
- exchanges/
- exchanges/#ex-sec
- exchanges/#all (and other options)
- exchanges/#otc-traders
- exchanges/#pp-sec

Why do some of them have "-sec" and others do not? We should make sure the URLs look good and are self-explanatory. This is especially important when shared on social media.

Second, when I navigate from Payment Processors to OTC Traders, I see the Payment Processors included in the OTC Traders list. I assume this behavior is not intended, because when I navigate from Exchanges to OTC Traders I do not see the Payment Processors included.

**Comment by Jujhar** [comment]
Yes that is the latest one. 

Updated as follows:
* exchanges/
* exchanges/#exchanges
* exchanges/#all (and other options)
* exchanges/#otc-traders
* exchanges/#payment-processors

Don't follow what you mean. The only unexpected thing is that if you load the page without clicking any it loads all three on one page but once you select one it should just show the one you selected.

**Comment by dajohi** [comment]
Closing due to "too many hardcoded sizing values in the CSS" and staleness.

**Pull request closed without merging**